account_payment_sale/wizard/sale_make_invoice_advance.py

```diff
@@ -7,10 +7,8 @@
 class SaleAdvancePaymentInv(models.TransientModel):
     _inherit = "sale.advance.payment.inv"
 
-    def _create_invoice(self, order, so_line, amount):
+    def _prepare_invoice_values(self, order, name, amount, so_line):
         """Copy payment mode from sale order to invoice"""
-        inv = super()._create_invoice(order, so_line, amount)
-        vals = order._get_payment_mode_vals({})
-        if vals:
-            inv.write(vals)
-        return inv
+        vals = super()._prepare_invoice_values(order, name, amount, so_line)
+        order._get_payment_mode_vals(vals)
```

Comment:
Why this double thing? Isn't enough with the other?

Reply:
I don't understand your remark. The code seems quite obvious and clear to me.